plasmapy/analytic/parameters.py

```diff
@@ -0,0 +1,852 @@
+"""Functions to calculate plasma parameters."""
+
+from astropy import units as u
+
+from astropy.units import (UnitConversionError, UnitsError, quantity_input,
+                           Quantity)
+
+from ..constants import (m_p, m_e, c, mu0, k_B, e, eps0, pi, ion_mass,
+                         charge_state)
+
+import numpy as np
+
+
+"""
+Values should be returned as an Astropy Quantity in SI units.
+
+If a quantity has several names, then the function name should be
+the one that provides the most physical insight into what the
+quantity represents.  For example, 'gyrofrequency' indicates
+gyration, while Larmor frequency indicates that this frequency was
+discovered by Larmor.
+
+The docstrings for plasma parameter methods should define these
+quantities in ways that are understandable to students who are
+taking their first course in plasma physics and yet are useful to
+experienced plasma physicists.
+
+Unit conversions involving angles must be treated with care.  Angles
+are dimensionless but do have units.  Angular velocity is often
+given in units of radians per second, though dimensionally this is
+equivalent to inverse seconds.  Astropy will treat radians
+dimensionlessly when using the dimensionless_angles() equivalency,
+but dimensionless_angles() does not account for multiplicative
+factor of 2*pi that is used when converting between frequency (1 /
+s) and angular velocity (rad / s).  A clear way to do this
+conversion is to set up an equivalency between cycles/s and Hz:
+
+  >>> f_ce = omega_ce.to(u.Hz, equivalencies=[(u.cy/u.s, u.Hz)])
+
+However, dimensionless_angles() does work when dividing a velocity
+by an angular frequency to get a length scale:
+
+  >>> d_i = (c/omega_pi).to(u.m, equivalencies=dimensionless_angles())
+
+"""
+
+
+def Alfven_speed(B, density, ion="p"):
+    """Returns the Alfven speed.
+
+    Parameters
+    ----------
+    B : Quantity
+        The magnetic field in units such as Tesla or Gauss
```

Comment:
This is probably obvious but I don't think it'd hurt to say
> The magnetic field **magnitude** in units such as Tesla or Gauss

We'll have to be careful attaching this to the Plasma class, as that one's B variable is a vector field.